src/js/media/html5.js

```diff
@@ -324,7 +324,7 @@ vjs.Html5.nativeSourceHandler.canHandleSource = function(source){
     return canPlayType(source.type);
   } else if (source.src) {
     // If no type, fall back to checking 'video/[EXTENSION]'
-    match = source.src.match(/\.([^\/\?]+)(\?[^\/]+)?$/i);
+    match = source.src.match(/\.([^.\/\?]+)(\?[^\/]+)?$/i);
```

Comment:
I was thinking the period needed to be escaped, but [apparently not](http://stackoverflow.com/questions/10397968/escape-dot-in-a-regex-range).